flyway-community-db-support/src/main/java/org/flywaydb/community/database/clickhouse/ClickHouseConfigurationExtension.java

```diff
@@ -0,0 +1,27 @@
+package org.flywaydb.community.database.clickhouse;
+
+import lombok.Getter;
+import org.flywaydb.core.extensibility.ConfigurationExtension;
+
+import java.util.Map;
+
+@Getter
+public class ClickHouseConfigurationExtension implements ConfigurationExtension {
+    private static final String CLUSTER_NAME = "flyway.clickhouse.clusterName";
+
+    private String clusterName;
+
+    @Override
+    public void extractParametersFromConfiguration(Map<String, String> configuration) {
+        clusterName = configuration.get(CLUSTER_NAME);
```

Comment:
```java
String clusterName = configuration.remove(CLUSTER_NAME);
if (clusterName != null) {
    this.clusterName = clusterName;
}
```
I think need to be like this. 
This solution was tested by me with spring boot flyway autoconfiguration. This method was calling multiple times during spring context initialization

Reply:
Do you have several Flyway configurations in your app? Each flyway extension (plugin) expects to be configured only once. PostgreSQLConfigurationExtension has the same issue if it is reconfigured. I fixed that for ClickHouse, but it looks like some problem with app configuration.